src/Microsoft.AspNet.Mvc.ModelBinding/ModelStateDictionary.cs

```diff
@@ -135,7 +135,10 @@ public ICollection<ModelState> Values
         /// <inheritdoc />
         public bool IsValid
         {
-            get { return ValidationState == ModelValidationState.Valid; }
+            get
+            {
+                return ErrorCount == 0;
+            }
```

Comment:
hmm, I'm not sure about this one. The purpose of this was to not report something as valid if validation hadn't run yet.